sdk/keyvault/internal/challenge_policy.go

```diff
@@ -27,18 +27,29 @@ import (
 const headerAuthorization = "Authorization"
 const bearerHeader = "Bearer "
 
+type KeyVaultChallengePolicyOptions struct {
+	// DisableChallengeResourceVerification controls whether the policy requires the
+	// authentication challenge resource to match the Key Vault or Managed HSM domain
+	DisableChallengeResourceVerification bool
+}
+
 type KeyVaultChallengePolicy struct {
 	// mainResource is the resource to be retrieved using the tenant specified in the credential
-	mainResource *temporal.Resource[azcore.AccessToken, acquiringResourceState]
-	cred         azcore.TokenCredential
-	scope        *string
-	tenantID     *string
+	mainResource            *temporal.Resource[azcore.AccessToken, acquiringResourceState]
+	cred                    azcore.TokenCredential
+	scope                   *string
+	tenantID                *string
+	verifyChallengeResource bool
```

Comment:
I suggest dropping the *Resource* postfix.  We have that on mainResource which is a Resource.  In this case this is a switch and not a resource.

```suggestion
	verifyChallenge bool
```

Reply:
We are specifically verifying the `resource` in the challenge, not the entire challenge itself. Dropping "resource" is misleading.